lib/skate/open_route_service_api/client.ex

```diff
@@ -0,0 +1,90 @@
+defmodule Skate.OpenRouteServiceAPI.Client do
+  @moduledoc """
+  An HTTP Client that reaches out to Open Route Service
+  """
+
+  alias Skate.OpenRouteServiceAPI.DirectionsRequest
+
+  @callback get_directions(DirectionsRequest.t()) :: {:ok, map()} | {:error, any()}
+
+  @doc """
+  Sends `request` to the OpenRouteService API and then sends the response through
+  `parse_response/1` before returning it
+  """
+  @spec get_directions(DirectionsRequest.t()) :: {:ok, map()} | {:error, any()}
```

Comment:
**question:** I think you can just do `@impl true` rather than repeating the typespec from the `@callback`.

Reply:
Ahh, okay. Yeah if changing it would make the docs worse then I'm fine leaving it as-is.